crates/ruff_linter/src/noqa.rs

```diff
@@ -424,21 +425,23 @@ impl<'a> FileNoqaDirectives<'a> {
 /// An individual file-level exemption (e.g., `# ruff: noqa` or `# ruff: noqa: F401, F841`). Like
 /// [`FileNoqaDirectives`], but only for a single line, as opposed to an aggregated set of exemptions
 /// across a source file.
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug)]
 pub(crate) enum ParsedFileExemption<'a> {
     /// The file-level exemption ignores all rules (e.g., `# ruff: noqa`).
     All,
     /// The file-level exemption ignores specific rules (e.g., `# ruff: noqa: F401, F841`).
-    Codes(Vec<&'a str>),
+    Codes(Codes<'a>),
 }
 
 impl<'a> ParsedFileExemption<'a> {
     /// Return a [`ParsedFileExemption`] for a given comment line.
-    fn try_extract(line: &'a str) -> Result<Option<Self>, ParseError> {
+    fn try_extract(line: &'a str, offset: TextSize) -> Result<Option<Self>, ParseError> {
```

Comment:
Nit: The current signature is ambigious in the sense that it's unclear whether `offset` is an offset relative to `line` (that's what I would assume) or if it's an absolute index into the source text. 

I suggest changingt the signature to `try_extract(comment_range: TextRange, source: &'a str)` to make this clear (we commonly use `source` to refer to the entire source text).

Reply:
Ah that makes sense. I was just trying to copy the signature of [Directive::try_extract](https://github.com/astral-sh/ruff/blob/c40b37aa36135283db0b4f16d4582d6879e43cf5/crates/ruff_linter/src/noqa.rs#L56), but this makes more sense here.